crates/outline_panel/src/outline_panel.rs

```diff
@@ -2149,11 +2218,24 @@ impl OutlinePanel {
             _ => false,
         };
 
-        let icon = if self.is_singleton_active(cx) {
-            None
+        // Check if this outline has children by looking at the next entries
+        let has_children = self.has_outline_children(outline, cx);
```

Comment:
It's not superb that we have to use this method during __rendering__ , that might happen fast, e.g. 120 times/second: and we loop over outlines in a potentially large file.

What's more odd, is that we call `render_outline(` in a loop over every item to render, so should know whether the outline has children or not.

Seems that we can rewrite it into something better.